alicloud/data_source_alicloud_market_product.go

```diff
@@ -0,0 +1,139 @@
+package alicloud
+
+import (
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/market"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func dataSourceAlicloudProduct() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudProductRead,
+
+		Schema: map[string]*schema.Schema{
+			"product_code": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			// Computed values.
+			"product": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"code": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"skus": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"sku_code": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"sku_name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"package_versions": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												// Query package version of API products is now available. Other products are to be provided by back-end services.
+												"package_version": {
```

Comment:
// Currently, the API products can return package_version, but others can not for ensure.